RELEASE/scripts/autoscend/auto_boris.ash

```diff
@@ -275,13 +275,73 @@ void boris_buySkills()
 	set_property("auto_borisSkills", my_level());
 }
 
+void borisDemandSandwich()
+{
+	//Boris can summon a sandwich 3 times per day at cost of 5 MP.
+	if(!in_boris())
+	{
+		return;
+	}
+	if(get_property("_demandSandwich").to_int() > 2) 		//max 3 casts a day
+	{
+		return;
+	}
+	
+	//use ongoing MP recovery to summon sandwiches as boris if you can get the best sandwich.
+	if(my_level() > 8)
+	{
+		while(my_mp() > 4 && get_property("_demandSandwich").to_int() < 3)
+		{
+			use_skill(1, $skill[Demand Sandwich]);
+		}
+	}
+	//if your level is too low for the best sandwich, summon a single inferior sandwich to tide you over when low on adventures and if you don't already have one in inventory.
+	//this part semi relies on a future update to consumption code to make it eat one item at a time.
+	else if(my_adventures() < 8 && item_amount($item[club sandwich]) == 0 && item_amount($item[PB&BP]) == 0)
+	{
+		if(my_mp() > 4 && get_property("_demandSandwich").to_int() < 3)
+		{
+			use_skill(1, $skill[Demand Sandwich]);
+		}
+	}
+}
+
+void borisOngoingLaugh()
+{
+	//ongoing conversion of MP to HP to be done after every adventure to prevent wasting MP regen from clancy.
+	//Laugh it off costs 1 MP to cast and gives either 1 or 2 HP randomly.
+	if(!in_boris())
+	{
+		return;
+	}
+	
+	//don't waste HP on convert, keep at least 10 MP and at least 50% of MP.
+	while((my_hp()+1) < my_maxhp() && my_mp() > 10 && my_mp() > (my_maxmp()/2))
+	{
+		//multi use without risking wastage.
+		int missingHP = my_maxhp() - my_hp();
+		int availableMP = my_mp() - (my_maxmp()/2);
+		if(my_mp() > 10)
+		{
+			availableMP = min(availableMP, (my_mp() - 10));
+		}
+		int castAmount = min(availableMP, (missingHP / 2));
+		use_skill(castAmount, $skill[Laugh it Off]);
+	}
+}
+
 boolean LM_boris()
 {
+	//this function is called early once every loop of doTasks() in autoscend.ash
+	//if something in this function returns true then it will restart the loop and get called again.
+	
 	if(!in_boris())
 	{
 		return false;
 	}
-
+	
+	borisDemandSandwich();
```

Comment:
So this is a good example of why it's helpful for `borisDemandSandwitch()`, etc. to return `boolean`. These calls are currently done early every `doTasks()` loop, but that might not always be true. What if someday we change `equipBaseline()` to take into account your current HP, and we heal to full via `borisOngoingLaugh()` after that, ending up with a stale baseline (e.g. running extra HP regen)?

If it's not a performance issue, why make the code harder to reason about?

Also, the call to `borisLaughItOff()` should just go into `acquireHP(int amnt)`, in case our equipment maximizer swaps out our equipment, either wasting heals or ensuring we're not at 100% for combat.

Reply:
>So this is a good example of why it's helpful for borisDemandSandwitch(), etc. to return boolean.

I looked several times and I am not seeing the reason. What is the reason we want borisDemandSandwich() to be a boolean? And what should its return values be?

>What if someday we change equipBaseline() to take into account your current HP, and we heal to full via borisOngoingLaugh() after that, ending up with a stale baseline (e.g. running extra HP regen)?

The laugh it off function I created is not designed to perform full heals. It is designed to be an ongoing conversion of excess MP regen from clancy into HP regen.

Losing HP to equipment change could cause some of it to be wasted, but it does that regardless. From testing I am seeing this function provide a massive improvement to boris surviveability as well as reducing the costs of healing. (because when it does do a full heal it needs to empty MP then fill it several times via galaktik MP resotrers, which are expensive)

Reply:
Actually... why does equipBaseline() exist?
Literally the only thing it does is
equipMaximizedGear();

Reply:
Made demand sandwich a boolean as request
renamed borislaughitoff and moved it to pre_adv file so it is called after equipbaseline()